### ISSUE TITLE
[methods.partition GridPartitioning] 

### ISSUE THREAD

**Issue by FarnazH** [theochem/DiverseSelector#134]
To do:

- [ ] Add test to reach 100% coverage
- [ ] Add comments in the code
- [ ] Add formulas to docstring and polish docstring (add reference).

**Comment by FanwangM** [comment]
The current GridPartition method is missing tests (commented in the corresponding tests). The problem is the usage of `compute_diversity` in 89f1d5d is outdated where the argument names does not work for `hypersphere_overlap_of_subset` in `compute_diversity` function. Once #138 is merged, hope we can solve this problem automatically. But if not, I will provide a quick fix myself.

**Comment by FarnazH** [comment]
**Post PR-#162:**

- **[Feature]** The same number of bins is used for partitioning each axis/dimension of feature space. It can be desirable to use different values for each axis. This extension should be easy to add.

**Comment by FarnazH** [comment]
@Ali-Tehrani, when putting together the `quick_start.ipynb` (see PR #186), I encountered the `RuntimeWarning: invalid value encountered in floor_divide bin_index = np.floor_divide(X - axis_minimum, bin_length)` from [L124 of selector/methods/partition.py](https://github.com/theochem/Selector/blob/main/selector/methods/partition.py#L124). The `bin_length` ends up being zero. I have copied the code snippet that reproduces this below. I didn't have time to look into it, can you please check what is going on?
 
```python
from sklearn.datasets import make_blobs
from selector.methods.partition import GridPartition

# generate n_sample data in 2D feature space forming 3 clusters
X, labels = make_blobs(n_samples=500, n_features=2, centers=2, random_state=42)

selector = GridPartition(numb_bins_axis=5, grid_method="equisized_dependent")
selected = selector.select(X, size=50, labels=labels)
print(len(selected))
```
